sqldelight-idea-plugin/src/main/kotlin/com/squareup/sqldelight/intellij/inspections/UnusedColumnInspection.kt

```diff
@@ -0,0 +1,136 @@
+import com.squareup.sqldelight.core.lang.util.prevSibling
+
+class UnusedColumnInspection : LocalInspectionTool() {
+
+  override fun buildVisitor(
+    holder: ProblemsHolder,
+    isOnTheFly: Boolean,
+    session: LocalInspectionToolSession
+  ): PsiElementVisitor = object : SqlVisitor() {
+    override fun visitCreateTableStmt(o: SqlCreateTableStmt) {
+      val candidates = o.findChildrenOfType<SqlColumnDef>()
+        .filter { columnDef ->
+          val columnName = columnDef.columnName
+          ReferencesSearch.search(columnName, columnName.useScope)
+            .allMatch { reference -> reference.element.parent is SqlColumnDef }
+        }
+
+      if (candidates.isEmpty()) {
+        return
+      }
+
+      val project = o.project
+      val psiManager = PsiManager.getInstance(project)
+      val files = FileTypeIndex.getFiles(SqlDelightFileType, GlobalSearchScope.allScope(project))
+      val hasSelectStar = files.asSequence()
+        .map { virtualFile -> psiManager.findFile(virtualFile) as SqlDelightFile }
+        .flatMap { file -> file.sqlStmtList?.stmtList.orEmpty() }
+        .filter { stmt ->
+          stmt.compoundSelectStmt != null && stmt.findChildrenOfType<SqlTableName>()
+            .any { name -> name.textMatches(o.tableName.text) }
+        }
+        .flatMap { stmt -> stmt.findChildrenOfType<SqlSelectStmt>() }
+        .flatMap { stmt -> stmt.resultColumnList }
+        .any { column -> column.textMatches("*") }
+      if (hasSelectStar) {
+        return
+      }
+
+      candidates.forEach { columnDef ->
+        holder.registerProblem(
+          columnDef.columnName, "Unused symbol", ProblemHighlightType.LIKE_UNUSED_SYMBOL,
+          SafeDeleteQuickFix(o, columnDef)
+        )
+      }
+    }
+  }
+
+  class SafeDeleteQuickFix(createTableStmt: SqlCreateTableStmt, columnDef: SqlColumnDef) :
+    LocalQuickFixOnPsiElement(createTableStmt) {
+
+    private val createTableRef = SmartPointerManager.getInstance(createTableStmt.project)
+      .createSmartPsiElementPointer(createTableStmt, createTableStmt.containingFile)
+
+    private val columnDefRef = SmartPointerManager.getInstance(columnDef.project)
+      .createSmartPsiElementPointer(columnDef, columnDef.containingFile)
+
+    override fun getFamilyName(): String = name
+
+    override fun getText(): String = "Safe delete ${columnDefRef.element?.columnName?.text.orEmpty()}"
+
+    override fun invoke(
+      project: Project,
+      file: PsiFile,
+      startElement: PsiElement,
+      endElement: PsiElement
+    ) {
+      WriteCommandAction.writeCommandAction(project).run<ReadOnlyModificationException> {
+        val createTable = createTableRef.element ?: return@run
+        val columnDef = columnDefRef.element ?: return@run
+
+        val columnDefString = createTable.columnDefList.filter { it !== columnDef }
+          .joinToString(",\n  ") { it.text.trim() }
+
+        val start = createTable.firstChild.nextSibling { it.elementType == SqlTypes.LP }
```

Comment:
Since you're only using the predicate for `elementType` I think you can remove the helper functions and instead do

```
val startTable = PsiTreeUtil.findSiblingForward(createTable.firstChild, SqlTypes.LP, null)
```

and similar for other callsites

Reply:
(Same for the usage in UnusedQueryInspector, I missed that one)

Reply:
> Since you're only using the predicate for `elementType` I think you can remove the helper functions and instead do
> 
> ```
> val startTable = PsiTreeUtil.findSiblingForward(createTable.firstChild, SqlTypes.LP, null)
> ```
> 
> and similar for other callsites

missed that one, thanks!